lib/node_modules/@stdlib/repl/lib/main.js

```diff
@@ -289,8 +301,14 @@ function REPL( options ) {
 	* @param {Object} key - key object
 	*/
 	function beforeKeypress( data, key ) {
-		previewCompleter.beforeKeypress( data, key );
-		ttyWrite.call( self._rli, data, key );
+		var settings = self._settings;
+		if ( key && key.name === 'backspace' && settings.autoDeletePairs ) {
+			self._autoCloser.beforeKeypress( data, key );
+		}
```

Comment:
@kgryte I think it would be better if we shift the `key.name === 'backspace'` check in the function itself as it's associated with autoCloser's logic and also limits the usage of the `beforeKeypress()` listener to just the case where the user enters backspace. What do you think?

Reply:
@Snehil-Shah Yes, agreed. That seems reasonable. Feel free to go ahead and refactor.